models/basket.py

```diff
@@ -201,12 +201,14 @@ def create_purchases(self):
                     line.tier.issue_instances(issue_count)
 
                     product = line.tier.parent
-                    if product.parent.type == "admissions":
-                        purchase_cls = AdmissionTicket
-                    elif product.parent.type in {"campervan", "parking"}:
-                        purchase_cls = Ticket
-                    else:
-                        purchase_cls = Purchase
+                    product_group_type = PRODUCT_GROUP_TYPES_DICT.get(
+                        product.parent.type
+                    )
+                    purchase_cls = (
+                        product_group_type.purchase_cls
+                        if product_group_type
+                        else Purchase
+                    )
```

Comment:
Could combine these two, by providing a dict lookup fallback/default?

```suggestion
                    purchase_cls = PRODUCT_GROUP_TYPES_DICT.get(
                        product.parent.type,
                        Purchase,
                    )
```

(although arguably less clear, partly due to the text wrapping to fit within line limits)

Reply:
Ah, nevermind; I see that there's an additional level of indirection for the `purchase_cls` lookup.  Hrm.  I'll ponder that a bit, but this seems OK as-is, and my suggestion's not valid.

Reply:
The problem is that the purchase class is an attribute on the thing in the dict, rather than directly being an element of it, so this doesn't work in and of itself.